### PR TITLE
feature/GIDS-69 | Allow entry value calculation inclusively in Invoice and other vouchers' modules

### PR DESCRIPTION
**Pull request by prateek-giddh**
* **What kind of change does this PR introduce?** (Bug fix, feature, docs update, ...)

* **What is the current behavior?** (You can also link to an open issue here)

* **What is the new behavior (if this is a feature change)?**

* **Other information**:

### CODE REVIEW

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html [hunk 1]
```diff
@@ -55,11 +55,12 @@
                             <input type="text" [(ngModel)]="currentTxn.inventory.unit.rate"
                                 [prefix]="selectedPrefixForCurrency" [suffix]="selectedSuffixForCurrency"
                                 [disabled]="shouldShowAdvanceReceiptMandatoryFields ? 'disabled' : null"
-                                [mask]="inputMaskFormat" (ngModelChange)="changePrice($event)" name=""
+                                [mask]="inputMaskFormat" [customDecimalPlaces]="4"
+                                (ngModelChange)="changePrice($event)" name=""
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html:59]: name attribute is empty

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts [hunk 2]
```diff
@@ -1010,11 +1010,19 @@ export class NewLedgerEntryPanelComponent implements OnInit, OnDestroy, OnChange
...
-    public calculateConversionRate(baseModel) {
+    /**
+     * Calculates conversion rate
+     *
+     * @param {*} baseModel Value to be converted
+     * @param {number} [customDecimalPaces] Optional custom decimal places (required for Rate as 4 digits are required for rate)
+     * @returns Converted rate
+     * @memberof NewLedgerEntryPanelComponent
+     */
+    public calculateConversionRate(baseModel, customDecimalPaces?: number) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.ts:1021]: Add return type

File: apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedger.vm.ts [hunk 3]
```diff
@@ -660,14 +660,22 @@ export class UpdateLedgerVm {
...
-    public calculateConversionRate(baseModel) {
+    /**
+     * Calculates conversion rate
+     *
+     * @param {*} baseModel Value to be converted
+     * @param {number} [customDecimalPaces] Optional custom decimal places (required for Rate as 4 digits are required for rate)
+     * @returns Converted rate
+     * @memberof UpdateLedgerVm
+     */
+    public calculateConversionRate(baseModel, customDecimalPaces?: number) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/ledger/components/updateLedgerEntryPanel/updateLedger.vm.ts:671]: Add return type

File: apps/web-giddh/src/app/shared/helpers/pipes/currencyPipe/currencyType.pipe.ts [hunk 4]
```diff
@@ -22,9 +22,9 @@ export class GiddhCurrencyPipe implements OnInit, OnDestroy, PipeTransform {
             this.store.select(p => p.settings.profile).pipe(takeUntil(this.destroyed$), distinctUntilKeyChanged('balanceDisplayFormat')).subscribe((o) => {
                 if (o && o.name) {
                     this._currencyNumberType = o.balanceDisplayFormat ? o.balanceDisplayFormat : 'IND_COMMA_SEPARATED';
-                    this._currencyDesimalType = o.balanceDecimalPlaces ? o.balanceDecimalPlaces : 0;
-                    if (this._currencyDesimalType) {
-                        localStorage.setItem('currencyDesimalType', this._currencyDesimalType.toString());
+                    this.currencyDecimalType = o.balanceDecimalPlaces ? o.balanceDecimalPlaces : 0;
+                    if (this.currencyDecimalType) {
+                        localStorage.setItem('currencyDesimalType', this.currencyDecimalType.toString());
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/helpers/pipes/currencyPipe/currencyType.pipe.ts:27]: correct spelling of decimal

File: apps/web-giddh/src/app/shared/helpers/pipes/currencyPipe/currencyType.pipe.ts [hunk 5]
```diff
@@ -49,20 +49,20 @@ export class GiddhCurrencyPipe implements OnInit, OnDestroy, PipeTransform {
         this.store.dispatch(this.settingsProfileActions.GetProfileInfo());
     }
 
-    public transform(input: number) {
+    public transform(input: number, customDecimalPlaces?: number) {
```
Comment (ravinderwalkover) [inline apps/web-giddh/src/app/shared/helpers/pipes/currencyPipe/currencyType.pipe.ts:52]: add return type

File: apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html [hunk 6]
```diff
@@ -55,11 +55,12 @@
...
                                 class="form-control text-right" />
                             <p class="mb-0 amnt-total" *ngIf="isLedgerAccountAllowsMultiCurrency">
                                 <ng-container *ngTemplateOutlet="totalWithCurrencySymbol"></ng-container>
-                                {{ currentTxn?.convertedRate | giddhCurrency}}
+                                {{ currentTxn?.convertedRate | giddhCurrency: 4}}
```
Comment (rohit-walkover) [inline apps/web-giddh/src/app/ledger/components/newLedgerEntryPanel/newLedgerEntryPanel.component.html:63]: please use constant for number pipe argument for '4'

**Pull request merged**